tests/action.test.ts

```diff
@@ -1370,6 +1370,119 @@ describe('Integration tests - action.ts', () => {
+`);
+
+    expect(mocks.TFError).not.toHaveBeenCalled();
+  });
+
+  test('Compose null test', async () => {
+    setDefaultInputs();
+
+    // Mock required Action inputs
+    // api_key - A testing farm server api key
+    vi.stubEnv('INPUT_API_KEY', 'abcdef-123456');
+    // git_url - An url to the GIT repository
+    vi.stubEnv(
+      'INPUT_GIT_URL',
+      'https://github.com/sclorg/testing-farm-as-github-action'
+    );
+    // tmt_plan_regex - A tmt plan regex which will be used for selecting plans. By default all plans are selected
+    vi.stubEnv('INPUT_TMT_PLAN_REGEX', 'fedora');
+
+    // Override default inputs
+    // compose - Set to empty string to simulate null/undefined input
+    vi.stubEnv('INPUT_COMPOSE', '');
+
+    // mock the pull request number and sha in the context
+    vi.stubEnv('INPUT_PR_NUMBER', '1');
+    vi.stubEnv('INPUT_COMMIT_SHA', 'd20d0c37d634a5303fa1e02edc9ea281897ba01a');
+
+    // Mock Testing Farm API
+    let capturedRequest: any;
+    vi.mocked(mocks.newRequest).mockImplementation(
+      async (request: NewRequest, _strict: boolean) => {
+        capturedRequest = request;
+        return Promise.resolve({
+          id: '1',
+          state: 'new',
+          created: '2021-08-24T14:15:22Z',
+          updated: '2021-08-24T14:15:22Z',
+        });
+      }
+    );
+    vi.mocked(mocks.requestDetails)
+      .mockResolvedValueOnce({
+        id: '1',
+        state: 'new',
+        result: null,
+        run_time: 1,
+        created: '2021-08-24T14:15:22Z',
+        updated: '2021-08-24T14:15:22Z',
+      })
+      .mockResolvedValueOnce({
+        id: '1',
+        state: 'queued',
+        result: null,
+        run_time: 61,
+        created: '2021-08-24T14:15:22Z',
+        updated: '2021-08-24T14:15:22Z',
+      })
+      .mockResolvedValueOnce({
+        id: '1',
+        state: 'pending',
+        result: null,
+        run_time: 121,
+        created: '2021-08-24T14:15:22Z',
+        updated: '2021-08-24T14:15:22Z',
+      })
+      .mockResolvedValueOnce({
+        id: '1',
+        state: 'running',
+        result: null,
+        run_time: 181,
+        created: '2021-08-24T14:15:22Z',
+        updated: '2021-08-24T14:15:22Z',
+      })
+      .mockResolvedValueOnce({
+        id: '1',
+        state: 'complete',
+        result: { overall: 'passed', summary: '\\o/' },
+        run_time: 3691,
+        created: '2021-08-24T14:15:22Z',
+        updated: '2021-08-24T14:15:22Z',
+      });
+
+    // Run action
+    const octokit = new Octokit({ auth: 'mock-token' });
+    const pr = await PullRequest.initialize(new CustomContext(), octokit);
+
+    await action(pr);
+
+    // hardware request has not been called
+    expect(mocks.unsafeNewRequest).not.toHaveBeenCalled();
```

Comment:
Please also check if we send null to TF

```suggestion
    expect(mocks.unsafeNewRequest).not.toHaveBeenCalled();
    
    expect(mocks.newRequest).toHaveBeenCalledWith(
      expect.objectContaining({
        environments: [
          expect.objectContaining({
            os: null, // os should be null
            arch: process.env['INPUT_ARCH'], // arch should be set
          }),
        ],
      }),
      false
    );
```